uui-components/src/overlays/Dropdown.tsx

```diff
@@ -153,17 +153,16 @@ export class Dropdown extends React.Component<DropdownProps, DropdownState> {
     }
 
     private handleMouseMove = (e: MouseEvent) => {
-        if (this.isInteractedOutside(e) && !this.isClientInArea(e)) { // User leave boundary area, close dropdown immediately or with this.props.closeDelay
-            this.clearCloseDropdownTimer();
+        if (this.isInteractedOutside(e) && this.isClientInArea(e) && !this.closeDropdownTimerId) { // User cursor in boundary area, but not inside toggler or body
             this.clearOpenDropdownTimer();
-
-            if (this.props.closeDelay) {
+            this.clearCloseDropdownTimer();
```

Comment:
It doesn't make sense to `clearCloseDropdownTimer` here, because we have a condition with  !this.closeDropdownTimerId in this branch